File: SDE_sheet/Arrays/05_sort_0_1_2/solution.rs
impl Solution {
    pub fn sort_colors(nums: &mut Vec<i32>) {
        let mut low = 0;
        let mut mid = 0;
        let mut high = nums.len() as i32 - 1;

        while mid as i32 <= high {
            match nums[mid] {
                0 => {
                    nums.swap(low, mid);
                    low += 1;
                    mid += 1;
                },
                1 => {
                    mid += 1;
                },
                2 => {
                    nums.swap(mid, high as usize);
                    high -= 1;
                },
                _ => {}
            }
        }
    }
}

File: SDE_sheet/Arrays/05_sort_0_1_2/solution.java
class Solution {
    public void sortColors(int[] nums) {
        int low = 0;
        int mid = 0;
        int high = nums.length - 1;

        while (mid <= high) {
            switch(nums[mid]) {
                case 0:
                    int tempLow = nums[low];
                    nums[low] = nums[mid];
                    nums[mid] = tempLow;
                    low++;
                    mid++;
                    break;
                case 1:
                    mid++;
                    break;
                case 2:
                    int tempHigh = nums[high];
                    nums[high] = nums[mid];
                    nums[mid] = tempHigh;
                    high--;
                    break;
            }
        }
    }
}

